src/nvim/api/ui.c

```diff
@@ -519,23 +517,48 @@ static void remote_ui_cmdline_show(UI *ui, Array args)
     ADD(new_item, copy_object(item.items[1]));
     ADD(new_contents, ARRAY_OBJ(new_item));
   }
-  ADD(new_args, ARRAY_OBJ(new_contents));
+  return new_contents;
+}
+
+static Array translate_firstarg(UI *ui, Array args)
+{
+  Array new_args = ARRAY_DICT_INIT;
+  Array contents = args.items[0].data.array;
+
+  ADD(new_args, ARRAY_OBJ(translate_contents(ui, contents)));
   for (size_t i = 1; i < args.size; i++) {
     ADD(new_args, copy_object(args.items[i]));
   }
-  push_call(ui, "cmdline_show", new_args);
+  return new_args;
 }
 
 static void remote_ui_event(UI *ui, char *name, Array args, bool *args_consumed)
 {
   if (!ui->ui_ext[kUINewgrid]) {
```

Comment:
Sure. "ext_grid_rev2" is not worse than anything else. The idea is also that UIs will in the long run typically not enable this extension per-se, as it will be implied by ext_multigrid or ext_float.

Once #8455, #6619 and #7466 are all merged it might also be worth declaring some kind of "revision" of the "smart ui protocol" (we could figure some better name), in form of a flag that enables everything so far. (to avoid flag mania, and to communicate some kind of message of "this is what an modern neovim UI looks like the end of 2018", or something)